Makefile

```diff
@@ -1,10 +1,51 @@
+.PHONY: build-pythonpackage dev-requirements format help release release-push \
+        release-tag release-unsafe requirements test test-format test-install \
+        test-lint test-pythonpackage test-types upgrade version docs
+
 .DEFAULT_GOAL := help
-.PHONY: docs
+
 SRC_DIRS = ./tutorralph
+PACKAGE=tutorralph
+PROJECT=tutor-contrib-ralph
+SOURCES=./setup.py ./$(PACKAGE)
+UPGRADE=CUSTOM_COMPILE_COMMAND='make upgrade' pip-compile --upgrade
 BLACK_OPTS = --exclude templates ${SRC_DIRS}
 
+###### Development
+
+COMMON_CONSTRAINTS_TXT=requirements/common_constraints.txt
+.PHONY: $(COMMON_CONSTRAINTS_TXT)
+$(COMMON_CONSTRAINTS_TXT):
+	wget -O "$(@)" https://raw.githubusercontent.com/edx/edx-lint/master/edx_lint/files/common_constraints.txt || touch "$(@)"
+
+upgrade: export CUSTOM_COMPILE_COMMAND=make upgrade
+upgrade: $(COMMON_CONSTRAINTS_TXT)
+	## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
+	pip install -qr requirements/pip-tools.txt
```

Comment:
did you try to generate the .txt files when just the .in files exist? if you do, this line will fail:

```
## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
pip install -qr requirements/pip-tools.txt
ERROR: Could not open requirements file: [Errno 2] No such file or directory: 'requirements/pip-tools.txt'
make: *** [Makefile:24: upgrade] Error 1
```

Reply:
Resolved